### PR TITLE
WIP: Add spell

### PR DESCRIPTION
**Pull request by hindrekhan**
This PR will add a spell casting feature. The spell can be cast using left mouse button. Currently there is one test spell called fireball. As of now there is no collision detection with spells.

### CODE REVIEW

File: spells/fireBall.lua [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+	local activeTime = 3
+	self.super:new(activeTime)
+
+	self.x = character.x
+	self.y = character.y
+	self.dir = character.rot
+	self.speed = 200
+
+	self.shader = love.graphics.newShader[[
+	extern number time;
```
Comment (jaanjah) [inline spells/fireBall.lua]: Perhaps put shaders in a new file and directory, maybe something like `shaders/fireball`

**Review by jaanjah — CHANGES_REQUESTED** [review]
What is the purpose of this file `spells/.fireBall.lua.swp`?

**Comment by jaanjah** [comment]
Add more info to PR body, something like `Add spell` can be a bit confusing, perhaps `Add spell feature`.
Also it's worth mentioning what spells this PR will add, is it just fireball or will there be more?

File: sceneManager.lua [hunk 2]
```diff
@@ -1,6 +1,9 @@
 require "scene"
 require "scenes/test"
 
+require "spell"
+require "spells/fireBall"
```
Comment (jaanjah) [inline sceneManager.lua]: You can just use `fireball`, it's 1 word. Make sure you also rename the files from `fireBall` to `fireball`

File: spells/fireball.lua [hunk 3]
```diff
@@ -0,0 +1,57 @@
...
+	self.speed = 200
+	self.shader = nil
+
+	local shaderPath
+    local shaderFile = io.open("./shaders/spells/fireball.glsl", "r")
+    if (shaderFile) then
+        shaderCode = shaderFile:read("*all")
+        shaderFile:close()
+
+		self.shader = love.graphics.newShader(shaderCode)
```
Comment (JaanJah) [inline spells/fireball.lua]: Use spaces here since you are using spaces for the rest of the code

**Review by jaanjah — APPROVED** [review]
Looks good! Some indentation problems, but that is a separate PR topic.

**Pull request merged**